src/irr/asset/CPLYMeshWriter.cpp

```diff
@@ -124,35 +131,36 @@ bool CPLYMeshWriter::writeAsset(io::IWriteFile* _file, const SAssetWriteParams&
             header += "property " + typeStr + " alpha\n";
         }
     }
-    if (desc->getMappedBuffer(asset::EVAI_ATTR2))
+    if (meshBuffer->getAttribBoundBuffer(2)->buffer)
     {
-        const asset::E_FORMAT t = desc->getAttribFormat(asset::EVAI_ATTR1);
+        const asset::E_FORMAT t = meshBuffer->getAttribFormat(1);
```

Comment:
this was a bug originally I think, you should be getting the format from index 2